modules/imgproc/src/filter.cpp

```diff
@@ -4670,9 +4671,11 @@ static bool dftFilter2D(int stype, int dtype, int kernel_type,
 
     Point anchor = Point(anchor_x, anchor_y);
     Mat kernel = Mat(Size(kernel_width, kernel_height), kernel_type, kernel_data, kernel_step);
+    Mat src = (src_step != dst_step) ? Mat(Size(width, height), stype, src_data, src_step) : Mat(Size(full_width-offset_x, full_height-offset_y), stype, src_data, src_step);
+    Mat dst = (src_step != dst_step) ? Mat(Size(width, height), dtype, dst_data, dst_step) : Mat(Size(full_width, full_height), dtype, dst_data, dst_step);
+    int _width  = (src_step != dst_step) ? width  : full_width;
+    int _height = (src_step != dst_step) ? height : full_height;
```

Comment:
> (src_step != dst_step)

I believe this condition should be improved (steps have no correlation with image sizes / ROIs).